### ISSUE TITLE
Beta Blog: Oracle UCP In OpenLiberty

### ISSUE THREAD

**Issue by alexsm82** [OpenLiberty/open-liberty#6571]
Epic #5867.

**Comment by alexsm82** [comment]
* Development progress update
a. Which Liberty feature does it relate to?
i. Human-readable name (eg WebSockets feature)
Java Database Connectivity
ii. Feature name (eg websockets-1.0)
jdbc-4.0, jdbc-4.1, jdbc-4.2, jdbc-4.3
iii. Is this a new feature or is it an update to an existing feature for this release?
This is an update to the existing JDBC features
iv. Provide the qualified name of all changed feature deliveries to this release? (eg: com.ibm.websphere.appserver.transportSecurity-1.0.esa)
Not sure what this is asking

b. Who is the target audience? eg application developers, devops pipeline devs, appserver administrators, etc 
application developers, devops, appserver administrators, database admins

c. Write a sentence or two that introduces it to someone new to the general technology/concept.
Support for Oracle Universal Connection Pool (UCP) has been added to Liberty in the 19.0.0.2 beta. Oracle UCP is a connection pool which can be used in place of Liberty's connection pool to take advantage of Oracle's high availability functionality.

d. What is new/changed in this release and why should they get excited about this new thing (2-3 sentences)? What was the problem before and how this new feature make their life easier? (Why should they care?)
Using Oracle UCP is now supported with Liberty in the 19.0.0.2 beta.  This will allow users of Oracle Real Application Clusters (RAC) to take advantage of Oracle’s high availability and performance functionality such as Fast Connection Failover (FCF), Fast Application Notification (FAN) and Oracle Notification Services (ONS).

e. Explain how to make it work. Include screenshots and/or code snippets to illustrate how to use it, if relevant. Provide server.xml snippet for how to enable the feature.

To use Oracle UCP with Liberty add the following server config to server.xml:
Add one of the JDBC features:
```
    <featureManager>
      <feature>jdbc-4.2</feature> <!-- Any of the other JDBC features could also be used -->
      <feature>jndi-1.0</feature> <!-- if JNDI lookup is desired -->
      <!-- Include any other features needed by your application -->
    </featureManager>
```

Add a library referencing the UCP jar and Oracle JDBC driver (both of which can be obtained from Oracle):

```
<library id="OracleUCPLib">
    <fileset dir="C:/Oracle/Drivers" includes="ojdbc8.jar ucp.jar"/>
</library>
```

Add a datasource using the new `properties.oracle.ucp` element:
```
<dataSource id="DefaultDataSource" jndiName="jdbc/oracleUCP" type="javax.sql.XADataSource">
    <jdbcDriver libraryRef="OracleUCPLib" />
    <containerAuthData user="user" password="pwd"/>
    <properties.oracle.ucp URL="jdbc:oracle:thin:@//localhost:1521/SAMPLEDB" connectionFactoryClassName="oracle.jdbc.xa.client.OracleXADataSource"/>
</dataSource>
```

Add any desired config (such as maxPoolSize or fastConnectionFailoverEnabled) to the `properties.oracle.ucp` element. It is important to note that when using Oracle UCP Liberty's connection pooling is disabled. As a result, the following data source properties are ignored: statementCacheSize and validationTimeout. The following connection manager properties are also ignored: agedTimeout, connectionTimeout, maxIdleTime, maxPoolSize, minPoolSize, purgePolicy, reapTime, maxConnectionsPerThread, maxConnectionsPerThreadLocal. Use the equivalent Oracle UCP functionality.

The standard JDBC APIs can be used to access the datasource.

f. Provide any public URLs where the reader can find out more about this specific update and/or about the wider technology.

Universal Connection Pool Developer's Guide:
https://docs.oracle.com/en/database/oracle/oracle-database/18/jjucp/index.html

g. Attach any images to this work item as .png files.

**Comment by dbarfield** [comment]
`iv. Provide the qualified name of all changed feature deliveries to this release? (eg: com.ibm.websphere.appserver.transportSecurity-1.0.esa)
Not sure what this is asking`

You say jdbc-4.0, jdbc-4.1, jdbc-4.2, jdbc-4.3 were changed then the answer above should be 
```com.ibm.websphere.appserver.jdbc-4.0.esa
com.ibm.websphere.appserver.jdbc-4.1.esa
com.ibm.websphere.appserver.jdbc-4.2.esa
com.ibm.websphere.appserver.jdbc-4.3.esa```

in this case it is obvious but it is not always so.

Could you explain why you (and another user) are having problems with that question ?  Is it the wording of the question ?

**Comment by alexsm82** [comment]
When I read that I wasn't entirely sure if it was asking which features were affected by the changes or which feature files were changed. I asked Nathan R (who I believe is the other person you are referring to) and he said he always put not sure and had never heard anything so that's what I did as well. Maybe updating the example to something like: "(eg: if you update the transport security feature - com.ibm.websphere.appserver.transportSecurity-1.0.esa)".
